db/schema/schema.sql

```diff
@@ -1,43 +1,51 @@
+-- to allow for the use of generating UUIDs
+CREATE EXTENSION
+IF NOT EXISTS pgcrypto;
+
 CREATE TABLE "user"
 (
-    "username" TEXT,
+    "id" UUID PRIMARY KEY DEFAULT gen_random_uuid(),
```

Comment:
Do I?  I think I might have at some point in the montagu db context perhaps?  There I think we used "natural primary keys"; so if you have a table of countries then ISO3 is a great PK because it's unique and unchanging. Following that through the db makes queries easier to follow (though one might look at montagu's queries and question if that was an overriding design idea).

I don't think that much is gained by adding an additional UUID column just to have a primary key if you're also going to put a unique constraint on a username (or email); it's just a layer of redirection that you never really benefit from.  If it will be important to rename the account this would not be the case, but most services don't support that, I feel

Reply:
Yes, that's how I recall the argument going! We definitely do want username to be unique. But yes, UUID would be useful in the user table if we do want to allow users to change their email addresses. Do we? Less likely that we'd want to change things like permission ids. And would have other options to migrate them if we did.

Reply:
I understand the logic but i disagree a little.
1.  Just seems a bit weird not having an ID as a PK in an SQL DB, ive never seen this before (except for composite keys).. 
2. 2. Secondly these need to be static and not change, username may be changed in the future and that would be a pain. Also if we ever have to merge DBs in the future or some migration conflicts could get nasty.. Thus is more robust to have these Ids...
3.  Also its a security issue it think because other users know the primary key to each user!!! not now but in the future when we allow other users to alter roles and things!!
4. Currently the Packit ids as UUIDS also so would be more consisant.. 

I don't see any drawbacks of using IDS, unless I'm missing anything?? It gives  future proofs at no cost